### PR TITLE
SQLite parser improvements

### PR DESCRIPTION
**Pull request by nickrum**
The first two commits should be self explanatory.

The third commit adds support for the `NULL` column constraint to the SQLite create table parser and compiler. This constraint is not mentioned in the SQLite documentation because it is a no-op. SQLite only parses it for compatibility reasons. As Knex sets this constraint on columns created with `.nullable()`, we have to parse it as well.
The alternativ would be to not set this constraint on nullable columns and to not parse it. Unfortunately, this would make altering columns on existing databases with nullable columns (and on those created with a raw query using the null constraint) impossible.

### CODE REVIEW

File: test/integration2/schema/alter.spec.js [hunk 1]
```diff
@@ -119,7 +122,25 @@ describe('Schema', () => {
...
+            });
+
+            it('removes an existing null constraint if a not null constraint is added to a column', async () => {
+              await knex.schema.alterTable('alter_table', (table) => {
+                table.boolean('column_nullable').notNullable().alter();
+              });
+
+              const tableAfter = (await knex.raw(QUERY_TABLE))[0].sql;
+
+              expect(
```
Comment (kibertoad) [inline test/integration2/schema/alter.spec.js:136]: this does not need to be awaited?

Comment (nickrum) [reply test/integration2/schema/alter.spec.js:136]: It definitely does! My bad, I think I misunderstood how the Chai Promise library works.

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Great job as usual!

**Comment by nickrum** [comment]
Thank you, I'm doing my best!

**Pull request merged**